HHS/Head-Start-TTADP
# [TTAHUB-2043] Fix duplication bug and add additional tests

## Description of change

There is a bug that causes personnel to be imported multiple times. This fixes the issue and ensures if the personnel is active with the current role it does NOT get created. Also added some tests.

**NOTE:** If a role has the same first and last name but the e-mail is different we expect two rows.


## How to test

- Review the code
- Review the script
- Make sure all test pass and we haven't missed any cases
- Double check the script correctly rolls up program personnel per (program id, grant id, first name, last name, role , email).
- Only 1 role should be active for the key, all other roles should **mapTo** the new column and be marked active = false.

You can use a query like this to drill into a few cases: 

SELECT
	"programId",
	"grantId",
	"firstName",
	"lastName",
	"role",
	count(id) as count
	FROM "ProgramPersonnel"
	GROUP BY 
	"programId",
	"grantId",
	"firstName",
	"lastName",
	"role"
	ORDER BY count(id) DESC

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-2043

## Checklists

### Every PR



### Production Deploy


### After merge/deploy
